packages/amplify_api_rest/android/src/main/kotlin/com/amazonaws/amplify/amplify_api_rest/AmplifyApiRestPlugin.kt

```diff
@@ -0,0 +1,311 @@
+  }
+
+  @VisibleForTesting
+  override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+    
+    if(call.method == "cancel"){
+      onCancel(result, (call.arguments as String))
+      return
+    }
+    
+    var arguments : Map<String, Any>
+
+    try {
+      arguments = checkArguments(call.arguments as Map<String, Any>)
+    }
+    catch(e: java.lang.Exception){
+      result.error("WrongMapInput", "The provided map input was not correct", e.message)
+      return
+    }
+    
+    when (call.method) {
+      "get" -> onGet(result, arguments)
+      "post" -> onPost(result, arguments)
+      "put" -> onPut(result, arguments)
+      "delete" -> onDelete(result, arguments)
+      else -> result.notImplemented()
+    }
+  }
+
+  private fun restFunctionHelper(
+          @NonNull flutterResult: Result,
+          @NonNull request: Map<String, *>,
+          @NonNull function3: KFunction3<
+                  @ParameterName(name = "restOptions") RestOptions,
+                  @ParameterName(name = "restConsumer") Consumer<RestResponse>,
+                  @ParameterName(name = "exceptionConsumer") Consumer<ApiException>,
+                  RestOperation?>,
+          @NonNull function4: KFunction4<
+                  @ParameterName(name = "apiName") String,
+                  @ParameterName(name = "restOptions") RestOptions,
+                  @ParameterName(name = "restConsumer") Consumer<RestResponse>,
+                  @ParameterName(name = "exceptionConsumer") Consumer<ApiException>,
+                  RestOperation?>
+  ){
+    if(FlutterRestInputs.validate(request)) {
+      val inputs = FlutterRestInputs(request)
+      val code = inputs.getCode()
+      val apiName: String? = inputs.getApiPath()
+      val options: RestOptions = FlutterRestInputs(request).getRestOptions()
+
+      try {
+
+        if(apiName == null){
+          var operation : RestOperation? = function3(options,
+                  Consumer { result -> prepareRestResponseResult(flutterResult, result, code) },
+                  Consumer { error -> prepareError(flutterResult, error, "hi", code) }
+          )
+          operationsMap[code] = operation!!
+        }
+        else{
+          var operation : RestOperation? = function4(
+                  apiName,
+                  options,
+                  Consumer { result -> prepareRestResponseResult(flutterResult, result, code) },
+                  Consumer { error -> prepareError(flutterResult, error, "hi", code) }
+          )
+          operationsMap[code] = operation!!
+        }
+
+      } catch(e: Exception) {
+        prepareError(flutterResult, e, "hi")
+      }
+    }
+    else{
+      prepareError(flutterResult, java.lang.Exception(FlutterAuthFailureMessage.MALFORMED.toString()), FlutterAuthFailureMessage.MALFORMED.toString())
+    }
+
+  }
+
+  private fun prepareRestResponseResult(@NonNull flutterResult: Result, @NonNull result: RestResponse, code: String = ""){
```

Comment:
Empty string is a dangerous default. Usually it means something has gone wrong, and you should be throwing an error, or using `null` to model "not present." Also, shouldn't a code be a number of some kind, not a string value?

Reply:
Sure we can use null instead as the default value. 

I use string because I'm using UUID.